### ISSUE TITLE
Error loading plugin 'fastlane-plugin-firebase_app_distribution': cannot load such file -- fastlane/plugin/firebase_app_distribution

### ISSUE THREAD

**Issue by RahathCherukuri** [fastlane/fastlane#16010]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

[✔] 🚀 
Unable to load fastlane-plugin-firebase_app_distribution plugin.

##### Command executed

  lane :distAll do
    commit = last_git_commit
    firebase_app_distribution(
      app: "APP-ID",
      groups: "dev, product, internal-qa",
      release_notes: commit[:message],
      firebase_cli_path: "/usr/local/bin/firebase"
    )
end

##### Complete output when running fastlane, including the stack trace and command used

<details>
[14:36:52]: Error loading plugin 'fastlane-plugin-firebase_app_distribution': cannot load such file -- fastlane/plugin/firebase_app_distribution
+-------------------------------------------+-----------+----------------------------------------------+
|                                             Used plugins                                             |
+-------------------------------------------+-----------+----------------------------------------------+
| Plugin                                    | Version   | Action                                       |
+-------------------------------------------+-----------+----------------------------------------------+
| fastlane-plugin-versioning                | 0.4.2     | get_version_number_from_git_branch           |
|                                           |           | increment_build_number_in_xcodeproj          |
|                                           |           | get_version_number_from_plist                |
|                                           |           | get_build_number_from_plist ci_build_number  |
|                                           |           | get_version_number_from_xcodeproj            |
|                                           |           | increment_version_number_in_xcodeproj        |
|                                           |           | increment_version_number_in_plist            |
|                                           |           | get_app_store_version_number                 |
|                                           |           | get_info_plist_path                          |
|                                           |           | get_build_number_from_xcodeproj              |
|                                           |           | increment_build_number_in_plist              |
| fastlane-plugin-firebase_app_distribution | undefined | No actions found                             |
+-------------------------------------------+-----------+----------------------------------------------+

[!] No actions were found while loading one or more plugins
    Please use `bundle exec fastlane` with plugins
    More info - https://docs.fastlane.tools/plugins/using-plugins/#run-with-plugins

[14:36:52]: ------------------------------
[14:36:52]: --- Step: default_platform ---
[14:36:52]: ------------------------------
[14:36:52]: Driving the lane 'ios distAll' 🚀
[14:36:52]: -----------------------------
[14:36:52]: --- Step: last_git_commit ---
[14:36:52]: -----------------------------
+------------------+-------------+
|          Lane Context          |
+------------------+-------------+
| DEFAULT_PLATFORM | ios         |
| PLATFORM_NAME    | ios         |
| LANE_NAME        | ios distAll |
+------------------+-------------+
[14:36:52]: Plugin 'firebase_app_distribution' was not properly loaded, make sure to follow the plugin docs for troubleshooting: https://docs.fastlane.tools/plugins/plugins-troubleshooting/

+------+------------------+-------------+
|           fastlane summary            |
+------+------------------+-------------+
| Step | Action           | Time (in s) |
+------+------------------+-------------+
| 1    | default_platform | 0           |
| 2    | last_git_commit  | 0           |
+------+------------------+-------------+

[14:36:52]: fastlane finished with errors

[!] Plugin 'firebase_app_distribution' was not properly loaded, make sure to follow the plugin docs for troubleshooting: https://docs.fastlane.tools/plugins/plugins-troubleshooting/

</details>

### Environment
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.1                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.21.1 (Apple Git-122.3)                           |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.15.1 (19B88)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 11.3.1                                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.141.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.0        |
| slack-notifier            | 2.3.2        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| public_suffix             | 2.0.5        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday                   | 0.17.3       |
| faraday_middleware        | 0.13.1       |
| gh_inspector              | 1.1.3        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| jwt                       | 2.1.0        |
| emoji_regex               | 1.0.1        |
| nanaimo                   | 0.2.6        |
| colored2                  | 3.1.2        |
| claide                    | 1.0.3        |
| CFPropertyList            | 3.0.2        |
| atomos                    | 0.1.3        |
| xcodeproj                 | 1.15.0       |
| unicode-display_width     | 1.6.1        |
| addressable               | 2.7.0        |
| tty-screen                | 0.7.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| babosa                    | 1.0.3        |
| excon                     | 0.72.0       |
| unf_ext                   | 0.0.7.6      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| fastimage                 | 2.1.7        |
| json                      | 2.3.0        |
| mini_magick               | 4.10.1       |
| dotenv                    | 2.7.5        |
| bundler                   | 2.1.4        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| httpclient                | 2.8.3        |
| multi_json                | 1.14.1       |
| signet                    | 0.12.0       |
| os                        | 1.0.1        |
| memoist                   | 0.16.2       |
| googleauth                | 0.10.0       |
| mini_mime                 | 1.0.2        |
| retriable                 | 3.1.2        |
| declarative-option        | 0.1.0        |
| declarative               | 0.0.10       |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| google-api-client         | 0.36.4       |
| digest-crc                | 0.4.1        |
| google-cloud-errors       | 1.0.0        |
| google-cloud-env          | 1.3.0        |
| google-cloud-core         | 1.5.0        |
| google-cloud-storage      | 1.25.1       |
| forwardable               | 1.2.0        |
| logger                    | 1.3.0        |
| stringio                  | 0.0.2        |
| ipaddr                    | 1.2.2        |
| openssl                   | 2.1.2        |
| ostruct                   | 0.1.0        |
| strscan                   | 1.0.0        |
| date                      | 2.0.0        |
| fileutils                 | 1.1.0        |
| etc                       | 1.0.1        |
| io-console                | 0.4.7        |
| zlib                      | 1.0.0        |
| libxml-ruby               | 3.1.0        |
| rexml                     | 3.1.9        |
| psych                     | 3.1.0        |
| mutex_m                   | 0.1.0        |
</details>

*generated on:* **2020-02-12**
</details>

**Comment by RahathCherukuri** [comment]
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.1                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.21.1 (Apple Git-122.3)                           |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.15.1 (19B88)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 11.3.1                                                         |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |


### fastlane files:

**No Fastfile found**


**No Appfile found**


### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.141.0 | ✅ Up-To-Date  |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.0        |
| slack-notifier            | 2.3.2        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| public_suffix             | 2.0.5        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday                   | 0.17.3       |
| faraday_middleware        | 0.13.1       |
| gh_inspector              | 1.1.3        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| jwt                       | 2.1.0        |
| emoji_regex               | 1.0.1        |
| nanaimo                   | 0.2.6        |
| colored2                  | 3.1.2        |
| claide                    | 1.0.3        |
| CFPropertyList            | 3.0.2        |
| atomos                    | 0.1.3        |
| xcodeproj                 | 1.15.0       |
| unicode-display_width     | 1.6.1        |
| addressable               | 2.7.0        |
| tty-screen                | 0.7.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| babosa                    | 1.0.3        |
| excon                     | 0.72.0       |
| unf_ext                   | 0.0.7.6      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| fastimage                 | 2.1.7        |
| json                      | 2.3.0        |
| mini_magick               | 4.10.1       |
| dotenv                    | 2.7.5        |
| bundler                   | 2.1.4        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| httpclient                | 2.8.3        |
| multi_json                | 1.14.1       |
| signet                    | 0.12.0       |
| os                        | 1.0.1        |
| memoist                   | 0.16.2       |
| googleauth                | 0.10.0       |
| mini_mime                 | 1.0.2        |
| retriable                 | 3.1.2        |
| declarative-option        | 0.1.0        |
| declarative               | 0.0.10       |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| google-api-client         | 0.36.4       |
| digest-crc                | 0.4.1        |
| google-cloud-errors       | 1.0.0        |
| google-cloud-env          | 1.3.0        |
| google-cloud-core         | 1.5.0        |
| google-cloud-storage      | 1.25.1       |
| forwardable               | 1.2.0        |
| logger                    | 1.3.0        |
| stringio                  | 0.0.2        |
| ipaddr                    | 1.2.2        |
| openssl                   | 2.1.2        |
| ostruct                   | 0.1.0        |
| strscan                   | 1.0.0        |
| date                      | 2.0.0        |
| fileutils                 | 1.1.0        |
| etc                       | 1.0.1        |
| io-console                | 0.4.7        |
| zlib                      | 1.0.0        |
| libxml-ruby               | 3.1.0        |
| rexml                     | 3.1.9        |
| psych                     | 3.1.0        |
| mutex_m                   | 0.1.0        |
</details>


*generated on:* **2020-02-12**
</details>

**Comment by rock-tran** [comment]
> <details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                                                                                |
| --------------------------- | -------------------------------------------------------------------------------------------------------------------- |
| OS                          | Linux freesky1102-pc 5.4.18-1-MANJARO #1 SMP PREEMPT Thu Feb 6 11:41:30 UTC 2020 x86_64 GNU/Linux                    |
| Ruby                        | 2.7.0                                                                                                                |
| Bundler?                    | false                                                                                                                |
| Git                         | git version 2.25.0                                                                                                   |
| Installation Source         | /usr/bin/fastlane                                                                                                    |
| Host                        | Linux freesky1102-pc 5.4.18-1-MANJARO #1 SMP PREEMPT Thu Feb 6 11:41:30 UTC 2020 x86_64 GNU/Linux (5.4.18-1-MANJARO) |
| Ruby Lib Dir                | /usr/lib                                                                                                             |
| OpenSSL Version             | OpenSSL 1.1.1d  10 Sep 2019                                                                                          |
| Is contained                | false                                                                                                                |
| Is homebrew                 | false                                                                                                                |
| Is installed via Fabric.app | false                                                                                                                |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |


### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#
# Maintainer: David Dang <david.dang@manadr.com>
# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:android)

platform :android do
  # Just use for local test
  before_all do
    ENV["SLACK_URL"] = ENV["ANDROID_SLACK_URL"]
  end

  desc "Runs all the tests"
  lane :test do
    gradle(task: "test")
  end

  desc "Submit a new Dev Build to Crashlytics Beta"
  lane :develop do
    # Using gradle to build this project

    versionName = File.read("../release_info/version_name.txt")
    releaseNote = ""

    isFromPR = ENV["CHANGE_ID"] ? true : false

    if isFromPR
        versionName += "-PR-#{ENV["CHANGE_ID"]}-#{ENV["BUILD_NUMBER"]}"
        releaseNote = ENV["CHANGE_TITLE"]
    else
        versionName += "-DEVELOP"
        releaseNote = "DEVELOP"
    end

    puts "VersionName = #{versionName}"
    puts "ReleaseNote = #{releaseNote}"

    versionCI = ENV["JOB_BASE_NAME"] ? ENV["JOB_BASE_NAME"] : "DEV"

    versionCode = File.read("../release_info/version_code.txt")

    gradle(
      task: "assemble",
      flavor: "dev",
      build_type: "Release",
    )

    firebase_app_distribution(
      app: APP_ID,
      firebase_cli_token: FIREBASE_CLI_TOKEN,
      groups: "QA",
      release_notes: releaseNote,
      apk_path: "app/build/outputs/apk/dev/release/app-dev-release.apk"
    )

    # Send message report to Slack
    slack(
       message: "Successfully deployed new Android Doctor App #{versionCI.upcase} version #{versionName} (#{versionCode}) to App Distribution :rocket:!",
    )

  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
json_key_file(ENV["GOOGLEPLAY_API_KEY_FILE"]) # Path to the json secret file - Follow https://docs.fastlane.tools/actions/supply/#setup to get one
package_name(APP_PACKAGE_NAME) # e.g. com.krausefx.app

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.141.0 | ✅ Up-To-Date  |


### Loaded fastlane plugins:

| Plugin                                    | Version   | Update-Status  |
| ----------------------------------------- | --------- | -------------- |
| fastlane-plugin-firebase_app_distribution | undefined | 💥 Check failed |


<details><summary><b>Loaded gems</b></summary>

| Gem                                       | Version      |
| ----------------------------------------- | ------------ |
| did_you_mean                              | 1.4.0        |
| slack-notifier                            | 2.3.2        |
| atomos                                    | 0.1.3        |
| CFPropertyList                            | 3.0.2        |
| claide                                    | 1.0.3        |
| colored2                                  | 3.1.2        |
| nanaimo                                   | 0.2.6        |
| xcodeproj                                 | 1.15.0       |
| rouge                                     | 2.0.7        |
| xcpretty                                  | 0.3.0        |
| terminal-notifier                         | 2.0.0        |
| unicode-display_width                     | 1.6.1        |
| terminal-table                            | 1.8.0        |
| plist                                     | 3.5.0        |
| addressable                               | 2.7.0        |
| multipart-post                            | 2.0.0        |
| word_wrap                                 | 1.0.0        |
| public_suffix                             | 2.0.5        |
| tty-screen                                | 0.7.1        |
| tty-cursor                                | 0.7.1        |
| tty-spinner                               | 0.9.3        |
| babosa                                    | 1.0.3        |
| colored                                   | 1.2          |
| highline                                  | 1.7.10       |
| commander-fastlane                        | 4.4.6        |
| excon                                     | 0.72.0       |
| faraday                                   | 0.17.3       |
| unf_ext                                   | 0.0.7.6      |
| unf                                       | 0.1.4        |
| domain_name                               | 0.5.20190701 |
| http-cookie                               | 1.0.3        |
| faraday-cookie_jar                        | 0.0.6        |
| faraday_middleware                        | 0.13.1       |
| fastimage                                 | 2.1.7        |
| gh_inspector                              | 1.1.3        |
| json                                      | 2.3.0        |
| mini_magick                               | 4.10.1       |
| multi_xml                                 | 0.6.0        |
| rubyzip                                   | 1.3.0        |
| security                                  | 0.1.3        |
| xcpretty-travis-formatter                 | 1.0.0        |
| dotenv                                    | 2.7.5        |
| bundler                                   | 2.1.4        |
| naturally                                 | 2.2.0        |
| simctl                                    | 1.6.8        |
| jwt                                       | 2.1.0        |
| uber                                      | 0.1.0        |
| declarative                               | 0.0.10       |
| declarative-option                        | 0.1.0        |
| representable                             | 3.0.4        |
| retriable                                 | 3.1.2        |
| mini_mime                                 | 1.0.2        |
| multi_json                                | 1.14.1       |
| signet                                    | 0.12.0       |
| memoist                                   | 0.16.2       |
| os                                        | 1.0.1        |
| googleauth                                | 0.10.0       |
| httpclient                                | 2.8.3        |
| google-api-client                         | 0.36.4       |
| google-cloud-env                          | 1.3.0        |
| google-cloud-errors                       | 1.0.0        |
| google-cloud-core                         | 1.5.0        |
| digest-crc                                | 0.4.1        |
| google-cloud-storage                      | 1.25.1       |
| emoji_regex                               | 1.0.1        |
| uri                                       | 0.10.0       |
| forwardable                               | 1.3.1        |
| logger                                    | 1.4.2        |
| cgi                                       | 0.1.0        |
| timeout                                   | 0.1.0        |
| stringio                                  | 0.1.0        |
| ipaddr                                    | 1.2.2        |
| openssl                                   | 2.1.2        |
| ostruct                                   | 0.2.0        |
| strscan                                   | 1.0.3        |
| date                                      | 3.0.0        |
| delegate                                  | 0.1.0        |
| fileutils                                 | 1.4.1        |
| etc                                       | 1.1.0        |
| io-console                                | 0.5.3        |
| zlib                                      | 1.1.0        |
| singleton                                 | 0.1.0        |
| rexml                                     | 3.2.3        |
| open3                                     | 0.1.0        |
| yaml                                      | 0.1.0        |
| psych                                     | 3.1.0        |
| mutex_m                                   | 0.1.0        |
| fastlane-plugin-firebase_app_distribution | 0.1.4        |
</details>


*generated on:* **2020-02-13**
</details>

The fastlane seems can't load plugin with normal user. However, if I run the above command with "sudo", it runs correctly.

**Comment by RahathCherukuri** [comment]
Yes with sudo it works fine.
"match" action its taking a long while when i run with sudo. So the issue now is i broke my build process into two steps:
1. Without sudo that does "match" and "gym". 
2. Once that is done now using sudo deploy("firebase_app_distribution").

**Comment by denysvarchenko** [comment]
I had the same issue and looks it's related to a Ruby version. So I have installed a rvm and downloaded Ruby 2.5.5. Then changed current Ruby version to 2.5.5 and called 'fastlane test' as usual

**Comment by RahathCherukuri** [comment]
ruby version 2.5.5 works. Thanks

**Comment by badredaha** [comment]
bundle exec fastlane fastlane update_plugins
